packages/core/test/providers/readonlyNetworksProvider.ts

```diff
@@ -0,0 +1,20 @@
+import { expect } from 'chai'
+import { getProvidersFromConfig } from '../../src/providers/network/readonlyNetworks/provider'
+import { Kovan, Mainnet, Rinkeby, Ropsten } from '../../src'
+import { JsonRpcProvider } from '@ethersproject/providers'
+
+describe('ReadonlyNetworksProvider', () => {
+  it('getProvidersFromConfig creates provider for each network that has URL', async () => {
+    const networks = [Mainnet, Ropsten, Rinkeby]
+    const urls = {
+      [Mainnet.chainId]: 'mainnetUrl',
+      [Rinkeby.chainId]: 'rinkebyUrl',
+      [Kovan.chainId]: 'kovanUrl',
+    }
+    const providers = getProvidersFromConfig(networks, urls)
```

Comment:
Should we throw error (or issue warning bleh 😩) when:
- urls is not subset of networks or
- don't represent the same set

Reply:
Or simply use `urls` alone, no need for networks